ambry-clustermap/src/main/java/com.github.ambry.clustermap/AmbryPartition.java

```diff
@@ -58,6 +60,21 @@
 
   @Override
   public PartitionState getPartitionState() {
+    // If there was a change to the sealed state of replicas in the cluster manager since the last check, refresh the
```

Comment:
`getWritablePartitions()` also does a liveness check and liveness changes are more frequent. Something similar can be done there as well however (but I think that is orthogonal to this patch).